src/MicroCellPosition.hpp

```diff
@@ -1,13 +1,23 @@
 #pragma once
 
+#include <stdexcept>
+
 #include "Direction.hpp"
 
-class MicroCellPosition {
-public:
+struct MicroCellPosition
+{
 	int x;
 	int y;
 
-	MicroCellPosition operator+(Direction direction) const;
-	MicroCellPosition& operator+=(Direction direction);
+	inline MicroCellPosition& operator+=(Direction direction)
+	{
+		switch (direction) {
+			case Right: this->x += 1; break;
+			case Up: this->y -= 1; break;
+			case Left: this->x -= 1; break;
+			case Down: this->y += 1; break;
+			default: throw std::out_of_range("direction");
```

Comment:
Exceptions and OpenMP don't play well together.  Convert this into an ERR_CRITICAL please.

Reply:
Done.